pynamodb/attributes.py

```diff
@@ -933,6 +933,22 @@ def validate(self, *, null_check: bool = False):
         return all(self.is_correctly_typed(k, v, null_check=null_check)
                    for k, v in self.get_attributes().items())
 
+    def _serialize_undeclared_attributes(self, values, container: Dict):
+        # Continue to serialize NULL values in "raw" map attributes for backwards compatibility.
+        # This special case behavior for "raw" attributes should be removed in the future.
+        for attr_name in values:
+            if attr_name not in self.get_attributes():
```

Comment:
Note that this is a new branch that should always return `True` for `MapAttribute` and sometimes return `True` for `DynamicMapAttribute`. Calling it out because there's some perf impact, though I think it should be marginal